Add initial electricity time series dataset (#3566)
huggingface/datasets: datasets/electricity/README.md

### Dataset Summary

This new dataset contains hourly kW electricity consumption time series of 370 Portuguese clients from 2011 to 2014.

### Supported Tasks and Leaderboards

[More Information Needed]

### Languages

[More Information Needed]

## Dataset Structure

Data set has no missing values. Values are in kW of each 15 min and are resampled to hourly frequency. 
Each column represent one client. Some clients were created after 2011. In these cases consumption were considered zero. All time labels report to Portuguese hour. However all days present 96 measures (24*4). Every year in March time change day (which has only 23 hours) the values between 1:00 am and 2:00 am are zero for all points. Every year in October time change day (which has 25 hours) the values between 1:00 am and 2:00 am aggregate the consumption of two hours.

### Data Instances

A sample from the training set is provided below:

```python
{
  'start': '2012-01-01 00:00:00', 
  'target': [14.0, 18.0, 21.0, 20.0, 22.0, 20.0, 20.0, 20.0, 13.0, 11.0], # <= this target value is a concatenated sample
  'feat_static_cat': [0], 
  'item_id': '0'
}
```

> We have two configurations `uci` and `lstnet`, which are specificed as as follows. 
> 
> The time series are resampled to hourly frequency. We test on 7 rolling windows of prediction lenght of 24. 
> 
> The `uci` validation therefore ends 24*7 time steps before the end of each time series. The training split ends 24 time steps before the end of the validation split. 
> 
> For the `lsnet` configuration we split the training window to be 0.6 of the full time series and the validation is the 0.8-th time series and the last 0.2 time windows are used as the test set of 7 rolling windows of the 24 time steps. Finally as in the LSTNet paper we only consider time series which are active in the year 2012-2014 which leaves us with 320 time series.

Reviewer: Some typos/grammer:

```suggestion
We have two configurations `uci` and `lstnet`, which are specificed as follows. 

The time series are resampled to hourly frequency. We test on 7 rolling windows of prediction length of 24. 

The `uci` validation therefore ends 24*7 time steps before the end of each time series. The training split ends 24 time steps before the end of the validation split. 

For the `lsnet` configuration we split the training window to be 0.6 of the full-time series and the validation is the 0.8-th time series and the last 0.2 time windows are used as the test set of 7 rolling windows of the 24 time steps. Finally, as in the LSTNet paper, we only consider time series that are active in the year 2012-2014, which leaves us with 320 time series.
```

### Data Fields

For this univariate regular time series we have:

> - `start`: a `datetime` as string of the first entry of each time serie in the dataset
> - `target`: an `array[float32]` of the actual target values
> 
> Given the `freq` and the `start` date, we can assign a datetime to each entry in the target array. Each time serie is also assigned a unique `feat_static_cat` integer id as well as an identifier `item_id`.

Reviewer: It's preferred to list and describe all the fields as follows: 

```suggestion
- `start`: a `datetime` as string of the first entry of each time serie in the dataset
- `target`: an `array[float32]` of the actual target values
- `feat_static_cat`: [field description]
- `item_id`: [field description]

Given the `freq` and the `start` date, we can assign a datetime to each entry in the target array.
```

(Also it would be nice if you could describe the `feat_stati_cat` field, which is a list of integer ids (you mention `integer id`), in more detail)
